Enumerable.rb

```diff
@@ -0,0 +1,165 @@
+module Enumerable
+    def my_each
+        return to_enum unless block_given?
+        n= self.length
+        x = 0
+        while x < n
+            yield(self[x])
+            x += 1
+        end
+         self
+    end
+    def my_each_with_index
+        return to_enum unless block_given?
+        n= self.length
+        x = 0
+        while x < n
+            yield(self[x], x)
+            x += 1
+        end
+         self
+    end
+    def my_select
+        return to_enum unless block_given?
+        pick = []
+        self.my_each { |x|
+            if yield(x)
+                pick.push(x)
+            end
+        }
+         pick
+    end
+    def my_all?
+        return to_enum unless block_given?
+        self.my_each { |x|
+        if yield(x) == false
+            return false
+        end
+        }
+         true
+    end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1, true, 'hi', []].my_all? != [1, true, 'hi', []].all?

[1,2,3,4,5,6].my_all?(Integer) != [1,2,3,4,5,6].all?(Integer)

['saheed', 'suretrust', 'oladele'].my_all?(/d/) != ['saheed', 'suretrust', 'oladele'].all?(/d/)

[1,2,3,4,5,6].my_all?(3) != [1,2,3,4,5,6].all?(3)
```